Final chapters of Fundamentals of Image and Imaging

bellabf opened the pull request:
Hello, everyone!

We are sending the imaging.mdx (Fundamentals of Imaging and its technical aspects of images) and the examples-preprocess.mdx (Preprocessing for Computer Vision).

These parts were written by @krmdel @klyap. There is also a small tutorial being written by @mfidabel with more examples which will come later on. I am just the one making the PR to help them out :smile: .

I will add the images once they are approved to the hf-course-assets page.

Best, Isabella

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/examples-preprocess.mdx:
```diff
@@ -0,0 +1,32 @@
+# Examples of pre-processing for Computer Vision
```
```suggestion
# Pre-processing for Computer Vision Tasks
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/examples-preprocess.mdx:
```diff
@@ -0,0 +1,32 @@
+# Examples of pre-processing for Computer Vision
+
+Now that we have seen what are images, how they are acquired and their impact, it is time to understand what are the operations we can perform them and how they are used during the model building process. 
```
```suggestion
Now that we have seen what are images, how they are acquired, and their impact, it is time to understand what operations we can perform and how they are used during the model-building process. 
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/examples-preprocess.mdx:
```diff
@@ -0,0 +1,32 @@
+- Mathematical
+- Transform operations.
+ 
+Each category encompasses specific techniques, such as Morphology under Logical Operations and Fourier and Principal Component Analysis (PCA) under Transforms. Understanding the distinction between elementwise and matrix operations is important in image manipulation. Element wise operations, such as raising an image to a power or dividing it by another image, involve processing each pixel individ ...
```
```suggestion
Each category encompasses different techniques, such as morphological operations under logical operations and fourier transforms and principal component analysis (PCA) under transforms. Understanding the distinction between element-wise and matrix operations is important in image manipulation. Element-wise operations, such as raising an image to a power or dividing it by another image, involve processing each pixel individually. This pixel-based approach contrasts with matrix operations, which utilize matrix theory for image manipulation. Having said that, you can do whatever you want with images, as they are matrices containing numbers!
```
Is morphology a technique though, isn't it more like a layer?

bellabf replied:
Hm, not in this context. Classically speaking, morphological operations are the group of techniques where you pass a structuring element in an image and normally the image stays about the same size. I think the classical example is erosion and dilation. In my experience, they are really common when you have the end users til interacting with the image or some human-in-the-loop. Sometimes the users are not happy about the segmentation mask (either because they are disconnected, or too small, or too big) so let them correct it using these morphological operations instead of trying to re-tune or re-run the model.

bellabf replied:
@krmdel @merveenoyan  I've added this clarification on text.

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/examples-preprocess.mdx:
```diff
@@ -0,0 +1,32 @@
+
+## Mathematical Tools in Image Processing
+
+Mathematical tools are indispensable in digital image processing. Set theory, for instance, is crucial for understanding and performing operations on images, particularly binary images. In these images, pixels are typically categorized as either foreground (1) or background (0). Operations such as union and intersection in set theory are used to determine relationships between features represente ...
```
```suggestion
Mathematical tools are indispensable in digital image processing. Set theory, for instance, is crucial for understanding and performing operations on images, particularly binary images. In these images, pixels are typically categorized as either foreground (1) or background (0). In set theory, operations such as union and intersection determine relationships between features represented by pixel coordinates. Intensity transformations and spatial filtering are other mathematical tools. They focus on manipulating pixel values within an image, where operators are applied to single images or a set of images for various purposes, like noise reduction.
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/examples-preprocess.mdx:
```diff
@@ -0,0 +1,32 @@
+Mathematical tools are indispensable in digital image processing. Set theory, for instance, is crucial for understanding and performing operations on images, particularly binary images. In these images, pixels are typically categorized as either foreground (1) or background (0). Operations such as union and intersection in set theory are used to determine relationships between features represente ...
+
+## Spatial Filtering Techniques and Image Enhancement
+Spatial filtering encompasses a broad range of applications in image processing, primarily focused on modifying images by altering each pixel's value based on its neighbouring pixels' values. Techniques include linear spatial filters, which can either blur (lowpass filters) or sharpen (high pass filters) an image. The properties and applications of different filter kernels, such as the Gaussian f ...
```
```suggestion
Spatial filtering encompasses a broad range of applications in image processing, primarily modifying images by altering each pixel's value based on its neighboring pixels' values. Techniques include linear spatial filters, which can blur (low pass filters) or sharpen (high pass filters) an image. The properties and applications of different filter kernels, such as the Gaussian and box filters, are contrasted. Sharpening filters emphasize transitions in intensity and are often implemented through digital differentiation techniques like the Laplacian, highlighting edges and discontinuities in an image.
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/examples-preprocess.mdx:
```diff
@@ -0,0 +1,32 @@
+Each category encompasses specific techniques, such as Morphology under Logical Operations and Fourier and Principal Component Analysis (PCA) under Transforms. Understanding the distinction between elementwise and matrix operations is important in image manipulation. Element wise operations, such as raising an image to a power or dividing it by another image, involve processing each pixel individ ...
+
+## Mathematical Tools in Image Processing
+
```
```suggestion
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/examples-preprocess.mdx:
```diff
@@ -0,0 +1,32 @@
+## Mathematical Tools in Image Processing
+
+Mathematical tools are indispensable in digital image processing. Set theory, for instance, is crucial for understanding and performing operations on images, particularly binary images. In these images, pixels are typically categorized as either foreground (1) or background (0). Operations such as union and intersection in set theory are used to determine relationships between features represente ...
+
```
Maybe you could also mention the most used transformations like contrast stretching or histogram matching

bellabf replied:
@krmdel this is an example of one where you can edit in the doc first and then change it on the .mdx file latter :)

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/imaging.mdx:
```diff
@@ -0,0 +1,44 @@
+Image acquisition in digital processing is the first step into turning the physical phenomena (what we see in real life), into a digital representation (what we see in our computers). It starts with the interaction between an illumination source and the subject being imaged. This illumination can be of various types, from conventional light sources to more sophisticated forms like electromagnetic ...
+
+## Sensor Technologies and Their Role in Image Acquisition
+As mentioned before, the first step into digital imaging are the sensors. To create a two-dimension image, single sensing elements (such as photodiodes) are moved along x and y axes. In contrast, the more common sensor strips capture images linearly in one direction. Thus, to obtain a complete 2D image, these strips are moved perpendicularly. This technology is commonly found in devices like flat ...
```
```suggestion
As mentioned before, the first step in digital imaging is the sensors. To create a two-dimensional image, single sensing elements (such as photodiodes) are moved along the x and y axes. In contrast, the more common sensor strips capture images linearly in one direction. Thus, to obtain a complete 2D image, these strips are moved perpendicularly. This technology is commonly found in devices like flatbed scanners and used in airborne imaging systems. In more specialized applications, like medical imaging (e.g., CAT scans), ring-configured sensor strips are used. These setups involve complex reconstruction advanced algorithms to transform the captured data into meaningful images. 
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/imaging.mdx:
```diff
@@ -0,0 +1,44 @@
+- Sampling, which digitizes the coordinate values
+- Quantization, which converts amplitude values into discrete quantities. 
+
+The resolution and quality of a digital image are significantly dependent on:
```
```suggestion
The resolution and quality of a digital image significantly depend on the following:
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/imaging.mdx:
```diff
@@ -0,0 +1,44 @@
+## Image Restoration and Reconstruction Techniques
+Image restoration focuses on recovering a degraded image using knowledge about the degradation phenomenon. It often involves modelling the degradation process and applying inverse processes to regain the original image.
+
+ In contrast, image enhancement is more subjective. It aims to improve the visual appearance of an image. Restoration techniques include dealing with issues like noise, which can originate from various sources during image acquisition or transmission. Advanced filters, both adaptive and non-adaptive, are used in this case because of their noise-reduction capabilities. In the field of medical imag ...
```
```suggestion
In contrast, image enhancement is more subjective. It aims to improve the visual appearance of an image. Restoration techniques include dealing with issues like noise, which can originate from various sources during image acquisition or transmission. Advanced filters, both adaptive and non-adaptive, are used in this case because of their noise-reduction capabilities. In medical imaging, particularly in computed tomography (CT), image reconstruction from projections is a crucial application.
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/imaging.mdx:
```diff
@@ -0,0 +1,44 @@
+ In contrast, image enhancement is more subjective. It aims to improve the visual appearance of an image. Restoration techniques include dealing with issues like noise, which can originate from various sources during image acquisition or transmission. Advanced filters, both adaptive and non-adaptive, are used in this case because of their noise-reduction capabilities. In the field of medical imag ...
+
+## Colour in Image Processing
+Colour is a powerful descriptor in image processing. It plays a role in object identification and recognition. Colour image processing includes both pseudo-colour and full-colour processing. 
```
```suggestion
Color is a powerful descriptor in image processing. It plays a role in object identification and recognition. Color image processing includes both pseudo-color and full-color processing. 
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/imaging.mdx:
```diff
@@ -0,0 +1,44 @@
+## Colour in Image Processing
+Colour is a powerful descriptor in image processing. It plays a role in object identification and recognition. Colour image processing includes both pseudo-colour and full-colour processing. 
+
+Pseudo-Colour processing assigns colours to grayscale intensities, while full-colour processing uses actual colour data from sensors. Understanding the fundamentals of colour, including human colour perception, the colour spectrum, and the attributes of chromatic light, is key. The fundamentals of colour involve the trichromatic nature of human vision, which perceives red, green, and blue. Colour ...
```
```suggestion
Pseudo-color processing assigns colors to grayscale intensities, while full-color processing uses actual color data from sensors. Understanding the fundamentals of color, including human color perception, the color spectrum, and the attributes of chromatic light, is key. The fundamentals of color involve the trichromatic nature of human vision, which perceives red, green, and blue. On the other hand, color perception is how the three types of cones in our eyes are stimulated. Finally, the color spectrum is the range of wavelengths in the electromagnetic spectrum that elicit distinct visual sensations.
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/imaging.mdx:
```diff
@@ -0,0 +1,44 @@
+
+Pseudo-Colour processing assigns colours to grayscale intensities, while full-colour processing uses actual colour data from sensors. Understanding the fundamentals of colour, including human colour perception, the colour spectrum, and the attributes of chromatic light, is key. The fundamentals of colour involve the trichromatic nature of human vision, which perceives red, green, and blue. Colour ...
+
+Different Colour models, such as RGB for monitors and cameras and CMY/CMYK for printing, standardize Colour representation in digital imaging. In the RGB Colour model, images have three components (i.e., channels), each for red, green, and blue. The pixel depth in RGB images determines the number of possible colours, with a typical full-colour image having a 24-bit depth (8 bits for each colour c ...
```
```suggestion
Different color models, such as RGB for monitors and cameras and CMY/CMYK for printing, standardize Colour representation in digital imaging. In the RGB Colour model, images have three components (i.e., channels), each for red, green, and blue. The pixel depth in RGB images determines the number of possible colors, with a typical full-color image having a 24-bit depth (8 bits for each color component). This allows for over 16 million possible colors! The RGB color cube represents the range of colors achievable in this model, with the grayscale extending from black to white. 
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/imaging.mdx:
```diff
@@ -0,0 +1,44 @@
+## Image Compression
+Data compression reduces the data needed to represent information. It distinguishes between data (the means of conveying information) and information itself. It targets redundancy, which is data that is either irrelevant or repetitive. For example, a 10:1 compression ratio indicates 90% data redundancy. 
+
+In digital image compression, particularly with 2-D intensity arrays, the three main types of redundancies are 
```
```suggestion
In digital image compression, particularly with 2-D intensity arrays, the three main types of redundancies are:
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/imaging.mdx:
```diff
@@ -0,0 +1,44 @@
+Data compression reduces the data needed to represent information. It distinguishes between data (the means of conveying information) and information itself. It targets redundancy, which is data that is either irrelevant or repetitive. For example, a 10:1 compression ratio indicates 90% data redundancy. 
+
+In digital image compression, particularly with 2-D intensity arrays, the three main types of redundancies are 
+- a) **Coding redundancy**. Coding redundancy is particularly prevalent in images where the distribution of intensity values does not spread evenly across all possible values, which is depicted as a non-uniform histogram. In such images, some intensity values occur more frequently than others, yet natural binary encoding assigns the same number of bits to represent each intensity value, regardles ...
```
```suggestion
- **Coding redundancy:** Coding redundancy is particularly prevalent in images where the distribution of intensity values does not spread evenly across all possible values, which is depicted as a non-uniform histogram. In such images, some intensity values occur more frequently than others, yet natural binary encoding assigns the same number of bits to represent each intensity value, regardless of its frequency. This means that common values are not encoded more efficiently than rare values, leading to inefficient use of bits and, thus coding redundancy. Ideally, more frequent values should be assigned shorter and less frequent values longer codes to minimize the number of bits used, which is not the case with natural binary encoding in non-uniform histograms.
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/imaging.mdx:
```diff
@@ -0,0 +1,44 @@
+
+In digital image compression, particularly with 2-D intensity arrays, the three main types of redundancies are 
+- a) **Coding redundancy**. Coding redundancy is particularly prevalent in images where the distribution of intensity values does not spread evenly across all possible values, which is depicted as a non-uniform histogram. In such images, some intensity values occur more frequently than others, yet natural binary encoding assigns the same number of bits to represent each intensity value, regardles ...
+- b) **Spatial and temporal redundancy**. Spatial and temporal redundancy appear in correlated pixel values within an image or across video frames.
```
```suggestion
- **Spatial and temporal redundancy:** Spatial and temporal redundancy appear in correlated pixel values within an image or across video frames.
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/imaging.mdx:
```diff
@@ -0,0 +1,44 @@
+In digital image compression, particularly with 2-D intensity arrays, the three main types of redundancies are 
+- a) **Coding redundancy**. Coding redundancy is particularly prevalent in images where the distribution of intensity values does not spread evenly across all possible values, which is depicted as a non-uniform histogram. In such images, some intensity values occur more frequently than others, yet natural binary encoding assigns the same number of bits to represent each intensity value, regardles ...
+- b) **Spatial and temporal redundancy**. Spatial and temporal redundancy appear in correlated pixel values within an image or across video frames.
+- c) **Irrelevant information**. Irrelevant information includes data ignored by the human visual system or unnecessary for the image's purpose.
```
```suggestion
- **Irrelevant information:** Irrelevant information includes data ignored by the human visual system or unnecessary for the image's purpose.
```

merveenoyan reviewed (commented):
Thank you so much 🤗

bellabf commented:
Hi everyone, 
I've addressed the reviewer's comments, so I could add the without having merge conflicts. :sweat_smile: 
@krmdel  could you address the comment that is left?

thanks everyone :hugs:

johko commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/examples-preprocess.mdx:
```diff
@@ -0,0 +1,31 @@
+- Mathematical
+- Transform operations.
+ 
+Each category encompasses different techniques, such as morphological operations under logical operations and fourier transforms and principal component analysis (PCA) under transforms. In this context, we refer to morphology as the group of operations tha use structuring elements to generate images of the same size by looking into the values of the pixel neighborhood. Understanding the distincti ...
```
getting a bit confused by the many "and"s in the first sentence, maybe it can be replaced as in my suggestion
```suggestion
Each category encompasses different techniques, such as morphological operations under logical operations or fourier transforms and principal component analysis (PCA) under transforms. In this context, we refer to morphology as the group of operations tha use structuring elements to generate images of the same size by looking into the values of the pixel neighborhood. Understanding the distinction between element-wise and matrix operations is important in image manipulation. Element-wise operations, such as raising an image to a power or dividing it by another image, involve processing each pixel individually. This pixel-based approach contrasts with matrix operations, which utilize matrix theory for image manipulation. Having said that, you can do whatever you want with images, as they are matrices containing numbers!
```

johko commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/imaging.mdx:
```diff
@@ -0,0 +1,67 @@
+
+Pseudo-color processing assigns colors to grayscale intensities, while full-color processing uses actual color data from sensors. Understanding the fundamentals of color, including human color perception, the color spectrum, and the attributes of chromatic light, is key. The fundamentals of color involve the trichromatic nature of human vision, which perceives red, green, and blue. On the other h ...
+
+Different color models, such as RGB for monitors and cameras and CMY/CMYK for printing, standardize Colour representation in digital imaging. In the RGB Colour model, images have three components (i.e., channels), each for red, green, and blue. The pixel depth in RGB images determines the number of possible colors, with a typical full-color image having a 24-bit depth (8 bits for each color compo ...
```
```suggestion
Different color models, such as RGB for monitors and cameras and CMY/CMYK for printing, standardize color representation in digital imaging. In the RGB color model, images have three components (i.e., channels), each for red, green, and blue. The pixel depth in RGB images determines the number of possible colors, with a typical full-color image having a 24-bit depth (8 bits for each color component). This allows for over 16 million possible colors! The RGB color cube represents the range of colors achievable in this model, with the grayscale extending from black to white. 
```

johko reviewed (approved):
Nice sections :+1:

merveenoyan reviewed (approved):
LGTM
